pyseir/rt/infer_rt.py

```diff
@@ -0,0 +1,617 @@
+        self.max_scaling_sigma = InferRtConstants.MAX_SCALING_OF_SIGMA
+        self.scale_sigma_from_count = InferRtConstants.SCALE_SIGMA_FROM_COUNT
+        self.tail_suppression_correction = InferRtConstants.TAIL_SUPPRESSION_CORRECTION
+        self.smooth_rt_map_composite = InferRtConstants.SMOOTH_RT_MAP_COMPOSITE
+        self.rt_smoothing_window_size = InferRtConstants.RT_SMOOTHING_WINDOW_SIZE
+        self.min_conf_width = InferRtConstants.MIN_CONF_WIDTH
+        self.log = structlog.getLogger(Rt_Inference_Target=self.display_name)
+        self.log_likelihood = None  # TODO: Add this later. Not in init.
+        self.log.info(event="Running:")
+
+    def get_timeseries(self, timeseries_type):
+        """
+        Given a timeseries type, return the dates, times, and requested values.
+
+        Parameters
+        ----------
+        timeseries_type: TimeseriesType
+            Which type of time-series to return.
+
+        Returns
+        -------
+        dates: list(datetime)
+            Dates for each observation
+        timeseries:
+            The requested timeseries.
+        """
+        timeseries_type = TimeseriesType(timeseries_type)
+
+        if timeseries_type is TimeseriesType.NEW_CASES:
+            return self.dates, self.cases
+        elif timeseries_type is TimeseriesType.NEW_DEATHS:
+            return self.dates, self.deaths
+        else:
+            raise ValueError
+
+    def evaluate_head_tail_suppression(self):
+        """
+        Evaluates how much time slows down (which suppresses Rt) as series approaches latest date
+        """
+        timeseries = pd.Series(1.0 * np.arange(0, 2 * self.window_size))
+        smoothed = timeseries.rolling(
+            self.window_size, win_type="gaussian", min_periods=self.kernel_std, center=True
+        ).mean(std=self.kernel_std)
+        delta = (smoothed - smoothed.shift()).tail(math.ceil(self.window_size / 2))
+
+        return delta[delta < 1.0]
+
+    def highest_density_interval(self, posteriors, ci):
+        """
+        Given a PMF, generate the confidence bands.
+
+        Parameters
+        ----------
+        posteriors: pd.DataFrame
+            Probability Mass Function to compute intervals for.
+        ci: float
+            Float confidence interval. Value of 0.95 will compute the upper and
+            lower bounds.
+
+        Returns
+        -------
+        ci_low: np.array
+            Low confidence intervals.
+        ci_high: np.array
+            High confidence intervals.
+        """
+        posterior_cdfs = posteriors.values.cumsum(axis=0)
+        low_idx_list = np.argmin(np.abs(posterior_cdfs - (1 - ci)), axis=0)
+        high_idx_list = np.argmin(np.abs(posterior_cdfs - ci), axis=0)
+        ci_low = self.r_list[low_idx_list]
+        ci_high = self.r_list[high_idx_list]
+        return ci_low, ci_high
+
+    def make_process_matrix(self, timeseries_scale=InferRtConstants.SCALE_SIGMA_FROM_COUNT):
```

Comment:
This isn't really accessing any instance specific variables other than constants, do you think this could be a function outside of the InferRt class?